client.go

```diff
@@ -54,13 +63,76 @@ func (a *authCreds) RequireTransportSecurity() bool {
 	return true
 }
 
+// Open creates a new Dgraph client by parsing a connection string of the form:
+// dgraph://<optional-login>:<optional-password>@<host>:<port>?<optional-params>
+// For example `dgraph://localhost:9080?sslmode=require`
+// It connects to the gRPC endpoint and, if credentials are provided, signs in the user.
+// An error can be returned if the Dgraph cluster is not yet ready to accept requests--the text
+// of the error in this case will contain the string "Please retry".
+func Open(connStr string) (*Dgraph, error) {
+	u, err := url.Parse(connStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid connection string: %w", err)
+	}
+
+	if !strings.HasPrefix(connStr, dgraphScheme) {
+		return nil, fmt.Errorf("invalid connection string: must start with %s", dgraphScheme)
+	}
+
+	opts := []ClientOption{}
+
+	apiKey := u.Query().Get(cloudAPIKeyParam)
+	sslMode := u.Query().Get(sslModeParam)
+
+	if apiKey != "" {
+		if len(apiKey) < 12 {
+			return nil, errors.New("invalid API key")
+		}
+		// Dgraph Cloud requires SSL
```

Comment:
again, we should just follow what the connection string says, we should not do this by default.